### DISCUSSION TITLE
Deploy crashes VPS

### DISCUSSION THREAD

**Discussion by Rehaeel** [coollabsio/coolify#2754] [Q&A] [answered]
Hi. 
I have an issue that after configuring the domain, GitHub app with ease, I tried to deploy my NextJS application and it crash the whole VPS. 

**VPS**: CPX11 on HETZNER (2vCPU, 2GB RAM as bare minimum). Tried also with Hostinger - 2vCPU & 8GB RAM, the same issue).
**Coolify Version**: latest v4.0.0-beta.306
**Next JS**: 13. Static site with default build commands (tried different approaches).

I was following all the instructions from your 1,5h guide on how to deploy Next JS app, but for every try, my VPS crash after few minutes and show blank screen of deploy logs:
![IMG_20240705_203247_914.jpg](https://github.com/coollabsio/coolify/assets/88146956/e0d704f4-c4e9-4ed6-ad23-644ea40d2130)

Every time it freezes on `npm i` command. I was trying to add the `--legacy-peer-deps` because it could get into dependencies issue, but I mapped the same setup on my locale machine and it threw no issues (matched the `node` and `npm` versions too):
| Locale | Coolify VPS |
| --- | --- |
| ![IMG_20240705_203254_015.jpg](https://github.com/coollabsio/coolify/assets/88146956/08e90092-87be-4c56-873a-4f9653c8f282) | ![IMG_20240705_203255_698.jpg](https://github.com/coollabsio/coolify/assets/88146956/d5fe5827-1818-42d4-9f01-1e2a60782494) |

Here are the screenshots on usage graphs (it went up to 200% 🫠)!: ![IMG_20240705_203250_762.jpg](https://github.com/coollabsio/coolify/assets/88146956/3d7497af-8982-4a8e-8e90-c9a292d62187)

Does anyone have an idea what might be the issue? The VPS matches the least requirements, and the app was not even deployed, so the only job that was running (except the Coolify itself) was the deploy job. 

Any help would be appreciated 🙏

**Comment by Rehaeel** [comment]
I tried with higher VPS tiers (this example **3vCPUs** * **8GB RAM,** and it worked, but hardly too. Went upto 200% and took so long that I thought it froze too.
<img width="1341" alt="image" src="https://github.com/coollabsio/coolify/assets/88146956/71165ecb-5b6b-4695-852f-b5991450524c">

**Comment by ikx94** [comment]
I'm having the same issue with a simple express API. How can we fix this? This has been happening for many months, which is why I can't use Coolify for serious projects.

**Comment by melang982** [comment]
Same issue, Nest.js, crashes on each third deployment or so. This is making me reconsider using Coolify. Hope it will get fixed.

**Comment by devcomputer** [comment]
Fails for me on debian12 ct in proxmox.

what i see in terminal:
`# curl -fsSL https://cdn.coollabs.io/coolify/install.sh | bash
Welcome to Coolify Installer!
This script will install everything for you. Sit back and relax.
Source code: https://github.com/coollabsio/coolify/blob/main/scripts/install.sh

---------------------------------------------
| Operating System  | debian 12
| Docker            | 27.0
| Coolify           | 4.0.0-beta.380
| Helper            | 1.0.4
| Realtime          | 1.0.5
---------------------------------------------

1. Installing required packages (curl, wget, git, jq, openssl). 
2. Check OpenSSH server configuration. 
 - OpenSSH server is installed.
 - SSH PermitRootLogin is enabled.
3. Check Docker Installation. 
 - Docker is installed.
4. Check Docker Configuration. 
 - Docker configuration is up to date.
5. Download required files from CDN. 
6. Make backup of .env to .env-20250108-180020
7. Propagating .env with new values - if necessary.
8. Checking for SSH key for localhost access.
 - Generating SSH key.
9. Installing Coolify (4.0.0-beta.380)
 - It could take a while based on your server's performance, network speed, stars, etc.
 - Please wait.
 - Until then, here's a joke for you:

If Bill Gates had a dime for every time Windows crashed ... Oh wait, he does.`

**Comment by rusakovic** [comment]
Confirm the issue. beta 390. Hetzner cpx31

**Comment by teknology** [comment]
I'm running the latest Coolify and running NextJS with a 2-core processor and 4GB memory without issues. Granted, running builds eat up CPU and memory, but that's expected, and Coolify has never crashed. I'm running swap, and I have nextjs running standalone mode. Note: The same Coolify server is also running Supabase for NextJS db. Plus the site is a real estate website pulling thousands of property listing data from MLS.

**Reply by Rehaeel** [reply]
dude, **`works on my machine`** is the worst answer. How can you help US that Coolify is not working? 😃

**Reply by teknology** [reply]
Did you miss where I said I have NextJS in standalone mode and I am running swap on the server? If you are not familiar with those terms, I recommend doing some Googling and educating yourself on the difference between standalone mode and standard mode as well as understanding what swap on a server is and what it's for. I'm speculating but Coolify probably isn't responding because this sounds like a NextJS configuration issue and not a Coolify issue. They have to deal with enough actual Coolify issues so it makes sense that this "issue" has no response. Another note: Coolify IMO Is a Docker wrapper where as it expands Docker with additional features. Docker can also use up a lot of memory if misconfigured.

**Comment by mmkostov** [comment]
Same error many months later. Seems like they're not serious about optimizing their app.

**Comment by andrasbacsai** [comment, marked as answer]
The building process requires CPU and memory, and totally done by Docker Engine. Coolify just tells Docker Engine how to build your application.

I suggest to enable swap (if it is not enabled already), and offload the building process to another server / CI like Github Actions. If your server crashes after this point, then please ping me on Discord and I will check what is going on your server.

**Reply by rusakovic** [reply]
Thank you.

Under the `I suggest to enable swap` do you mean this setting?

![image](https://github.com/user-attachments/assets/907d35fc-b4f1-4cb8-b02b-1db119ecac60)

2. Does  Coolify `and offload the building process to another server / CI like Github Actions.` support this function?
Or it's a custom implementation?

3. Is it possible to limit Deploy resource consumption on the same server? It's OK for me to deploy for a long time on 0.5 CPU, for example

**Reply by ghost** [reply]
Did you manage to find a solution to this?
Currently facing the same issue.

**Reply by rusakovic** [reply]
> Did you manage to find a solution to this? Currently facing the same issue.

I just build with Github actions. Coolify doesn't work.

**Reply by ghost** [reply]
Hi. Thank you for your response. I have fixed the issue by adding 1gb of swap space. You can try that too and let me know if you have fixed it!

**Reply by arshx86** [reply]
> Hi. Thank you for your response. I have fixed the issue by adding 1gb of swap space. You can try that too and let me know if you have fixed it!

can you tell me how exactly u did it

**Reply by ghost** [reply]
> > Hi. Thank you for your response. I have fixed the issue by adding 1gb of swap space. You can try that too and let me know if you have fixed it!
> 
> can you tell me how exactly u did it

https://www.digitalocean.com/community/tutorials/how-to-add-swap-space-on-ubuntu-20-04#step-4-enabling-the-swap-file

I followed this guide to add it.

**Reply by farukcan** [reply]
@xyztsx222 thank you so much. adding swap fixed the issue.
